src/lein_monolith/plugin.clj

```diff
@@ -42,6 +43,46 @@
     subprojects))
 
 
+(defn- with-meta*
+  "Returns an object of the same type and value as obj, with map m as its
+  metadata if the object can hold metadata."
+  [obj m]
+  (if (instance? clojure.lang.IObj obj)
+    (with-meta obj m)
+    obj))
+
+
+(defn- relativize
+  [root path]
+  (let [path-file (io/file path)]
+    (str (if (.isAbsolute path-file)
+           (.relativize (-> root io/file (.toPath))
+                        (.toPath path-file))
+           path))))
+
+
+(defn- relativize-path
+  [{:keys [root] :as project} key]
+  (cond (re-find #"-path$" (name key))
+        (update project key (partial relativize root))
+
+        (re-find #"-paths$" (name key))
+        (update project key #(with-meta* (map (partial relativize root) %)
+                               (meta %)))
+
+        :else project))
+
+
+(defn- relativize-paths
```

Comment:
This was copied from https://github.com/technomancy/leiningen/blob/master/leiningen-core/src/leiningen/core/project.clj#L482, but modified to relativize instead of absolutize.